internal/pkg/exec/docker.go

```diff
@@ -166,6 +166,36 @@ func (c DockerCommand) CheckDockerEngineRunning() error {
 	}
 }
 
+// GetPlatform will run the `docker version` command to get the OS/Arch.
+func (c DockerCommand) GetPlatform() (operatingSystem, architecture string, err error) {
+	var os string
+	var arch string
+	if _, err := exec.LookPath("docker"); err != nil {
+		return "", "", ErrDockerCommandNotFound
+	}
+	osBuf := &bytes.Buffer{}
+	err = c.runner.Run("docker", []string{"version", "-f", "'{{.Server.Os}}'"}, Stdout(osBuf))
```

Comment:
Stupid question: why do we use `Server` instead of `Client`?

Reply:
Good question! Because the user could be on a Windows machine running WSL, for example, in which case the host os/arch would be different than the engine os/arch, and it's the engine one we're interested in because that will match the container platform.